tests/Unit/NumericalAlgorithms/LinearOperators/Test_CoefficientTransforms.cpp

```diff
@@ -0,0 +1,162 @@
+        }
+      }
+
+      CHECK_ITERABLE_APPROX(u_expected, u_back);
+    };
+    helper(1000);
+    helper(order / 2);
+    if (order > 4) {
+      helper(order / 3);
+    }
+  }
+}
+
+void test_3d() noexcept {
+  for (size_t order = 3; order < 7; ++order) {
+    CAPTURE(order);
+    const Mesh<3> mesh({{order + 1, order, order - 1}},
+                       Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto);
+    const auto logical_coords = logical_coordinates(mesh);
+    const auto helper =
+        [&logical_coords, &mesh, order ](const size_t second_term) noexcept {
+      DataVector u_expected(mesh.number_of_grid_points());
+      for (size_t s = 0; s < mesh.number_of_grid_points(); ++s) {
+        u_expected[s] =
+            Spectral::compute_basis_function_value<Spectral::Basis::Legendre>(
+                order, get<0>(logical_coords)[s]) *
+            Spectral::compute_basis_function_value<Spectral::Basis::Legendre>(
+                order - 1, get<1>(logical_coords)[s]) *
+            Spectral::compute_basis_function_value<Spectral::Basis::Legendre>(
+                order - 2, get<2>(logical_coords)[s]);
+        if (second_term != 1000) {
+          u_expected[s] +=
+              Spectral::compute_basis_function_value<Spectral::Basis::Legendre>(
+                  second_term, get<0>(logical_coords)[s]) *
+              Spectral::compute_basis_function_value<Spectral::Basis::Legendre>(
+                  second_term, get<1>(logical_coords)[s]) *
+              Spectral::compute_basis_function_value<Spectral::Basis::Legendre>(
+                  second_term, get<2>(logical_coords)[s]);
+        }
+      }
+      const ModalVector u_coeffs =
+          nodal_to_modal_coefficients(u_expected, mesh);
+      const DataVector u_back = modal_to_nodal_coefficients(u_coeffs, mesh);
+      for (IndexIterator<3> index_it(mesh.extents()); index_it; ++index_it) {
+        if (((*index_it)[0] == order and (*index_it)[1] == order - 1 and
+             (*index_it)[2] == order - 2) or
+            ((*index_it)[0] == second_term and (*index_it)[1] == second_term and
+             (*index_it)[2] == second_term)) {
+          CHECK(u_coeffs[index_it.collapsed_index()] == approx(1.0));
+        } else {
+          CHECK(u_coeffs[index_it.collapsed_index()] == approx(0.0));
+        }
+      }
+
+      CHECK_ITERABLE_APPROX(u_expected, u_back);
+    };
+    helper(1000);
+    helper(order / 2);
+    if (order > 4) {
+      helper(order / 3);
+    }
+  }
+}
+}  // namespace test_legendre_polynomials
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Numerical.LinearOperators.CoefficientTransforms",
+                  "[NumericalAlgorithms][LinearOperators][Unit]") {
+  test_legendre_polynomials::test_1d();
+  test_legendre_polynomials::test_2d();
+  test_legendre_polynomials::test_3d();
+  // We can't test Chebyshev because our Chebyshev matrices implementation is
+  // incomplete
```

Comment:
What is missing? As far as I remember, `Chebyshev.cpp` implements everything that `Legendre.cpp` does.

Possibly, the only missing piece is to add a `Chebyshev` case in [Spectral.cpp:323](https://github.com/sxs-collaboration/spectre/blob/8aced90cb83652aa043ff04640f1d43593fc1229/src/NumericalAlgorithms/Spectral/Spectral.cpp#L323)

Reply:
From what I could tell the entire interface. I can't retrieve any of the matrices.
```
Node: 0 Proc: 0
Line: 349 of ../src/NumericalAlgorithms/Spectral/Spectral.cpp
Function: decltype(auto) Spectral::(anonymous namespace)::get_spectral_quantity_for_mesh(F &&, const Mesh<1> &) [F = (lambda at ../src/NumericalAlgorithms/Spectral/Spectral.cpp:369:1)]
Missing basis case for spectral quantity
```

Reply:
That should be solved by just adding that case to the place I linked above. It's where we translate from the runtime-info in `Mesh` to the compile-time interface of `Spectral

Reply:
Sure, but I have no idea in how many other places tests for Chebyshev need to be added then.

Reply:
I think this should be done carefully in a different PR since it's not high priority and I think requires careful thought to make sure everything ends up tested.